flink-python/pyflink/datastream/state.py

```diff
@@ -281,6 +283,110 @@ def __iter__(self) -> Iterator[K]:
         return iter(self.keys())
 
 
+class ReadOnlyBroadcastState(State, Generic[K, V]):
+    """
+    A read-only view of the :class:`BroadcastState`.
+    Although read-only, the user code should not modify the value returned by the :meth:`get` or the
+    items returned by :meth:`items`, as this can lead to inconsistent states. The reason for this is
+    that we do not create extra copies of the elements for performance reasons.
+    """
+
+    @abstractmethod
+    def get(self, key: K) -> V:
+        """
+        Returns the current value associated with the given key.
+        """
+        pass
+
+    @abstractmethod
+    def contains(self, key: K) -> bool:
+        """
+        Returns whether there exists the given mapping.
+        """
+        pass
+
+    @abstractmethod
+    def items(self) -> Iterable[Tuple[K, V]]:
+        """
+        Returns all the mappings in the state.
+        """
+        pass
+
+    @abstractmethod
+    def keys(self) -> Iterable[K]:
```

Comment:
The methods in this class are not the same as the methods in Java `ReadOnlyBroadcastState`. Is this intended? If so, could you explain it a bit? This class is a public class and so we need to be careful of it.

Reply:
We don't have the exactly same interfaces for `MapState`, which might be more intuitive for python users, so I just align `BroadcastState` with `MapState`.

Reply:
Since users already have `items()` interface, `keys()` and `values()` just making it more like `dict` in python, without not much extra code giving that the underlying state runtime implementation is actually map state too.